script/legacy/legacy40.php

```diff
@@ -294,29 +295,33 @@ function getKeyboardInfo($keyboard, $languageid) {
 
     //if(!empty($keyboard->MinKeymanWebVersion)) $jskeyboard['minVersion'] = $keyboard->MinKeymanWebVersion;
     //if(!empty($keyboard->MaxKeymanWebVersion)) $jskeyboard['maxVersion'] = $keyboard->MaxKeymanWebVersion;
-        
+
     $device_ios = KeyboardInfoPlatformSupportToLegacyDeviceTable($keyboard_info->platformSupport, 'ios');
     $device_android = KeyboardInfoPlatformSupportToLegacyDeviceTable($keyboard_info->platformSupport, 'android');
-    
+
     $jskeyboard['devices'] = array(
       'phone' => max($device_ios, $device_android),
       'tablet' => max($device_ios, $device_android),
       'desktop' => KeyboardInfoPlatformSupportToLegacyDeviceTable($keyboard_info->platformSupport, 'desktopWeb')
     );
-    
+
     // Load languages
+
     $jslanguages = array();
-    $languages = DB_LoadKeyboardLanguages($keyboard['keyboard_id']);
+    $languages = [];
+    foreach($allKeyboardLanguages as $akl) {
```

Comment:
Performance could probably be further improved with a one-time scan of this array to break it into per-keyboard, rather than scanning it for each keyboard O(n) rather than O(n2). Is it worth it? Probably...

Reply:
Yeah, it was worth it 😜